### PR TITLE
make atomics a shared library to fix build on android

### PR DESCRIPTION
**Pull request by michaelknoch**
This PR makes the latest version of swift-atomics compatible with our android build. They [added cmake support recently](https://github.com/flowkey/swift-atomics/commit/67bbf646d886d67ccda4d4ed1ca815efa83c7b39) which makes our CMakeLists.txt obsolete. Still I had to apply two tweaks to make the build work with our setup.

You think it makes sense to propose these changes to the official repo? @ephemer

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ cmake_minimum_required(VERSION 3.16)
 project(swift-atomics
   LANGUAGES C Swift)
 
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
```
Comment (michaelknoch) [inline CMakeLists.txt]: otherwise it tries to find `cmake/modules` in the root of the player directory

File: Sources/Atomics/CMakeLists.txt [hunk 2]
```diff
@@ -7,7 +7,7 @@ Licensed under Apache License v2.0 with Runtime Library Exception
 See https://swift.org/LICENSE.txt for license information
 #]]
 
-add_library(Atomics
+add_library(Atomics SHARED
```
Comment (michaelknoch) [inline Sources/Atomics/CMakeLists.txt:10]: linking Atomics statically leads to the following linker error:

```
ld.gold: error: lib/libAtomics.a: no archive symbol table (run ranlib)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
<unknown>:0: error: link command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

File: CMakeLists.txt (hunk 1 above)
Comment (michaelknoch) [reply CMakeLists.txt]: proposed this change to the official repo already
https://github.com/apple/swift-atomics/pull/52

File: Sources/Atomics/CMakeLists.txt (hunk 2 above)
Comment (ephemer) [reply Sources/Atomics/CMakeLists.txt:10]: Ok! Not 100% sure about that change in the long term: ideally we'd have "everything" linked statically in order to reduce final binary size. But our setup does not support that atm, so changing this to dynamic linking seems fine to me for now

File: CMakeLists.txt (hunk 1 above)
Comment (ephemer) [reply CMakeLists.txt]: LGTM

**Review by ephemer — APPROVED** [review]

**Comment by michaelknoch** [comment]
Since https://github.com/apple/swift-atomics/pull/52 was merged in the apple repository I was able to reduce the diff to the shared library thing only. I'll merge it into the branch `dynamic-library` which we would then reference from the android player.

fyi @ephemer

**Pull request merged**